runtime/Cpp/runtime/src/support/StringUtils.h

```diff
@@ -11,31 +11,33 @@ namespace antlrcpp {
   // For all conversions utf8 <-> utf32.
   // VS 2015 and VS 2017 have different bugs in std::codecvt_utf8<char32_t> (VS 2013 works fine).
 #if defined(_MSC_VER) && _MSC_VER >= 1900 && _MSC_VER < 2000
-  static std::wstring_convert<std::codecvt_utf8<__int32>, __int32> utfConverter;
+  using UtfConverterType = std::wstring_convert<std::codecvt_utf8<__int32>, __int32>;
+  using UtfConverterWide = std::u32string;
 #else
-  static std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> utfConverter;
+  using UtfConverterType = std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t>;
+  using UtfConverterWide = std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t>::wide_string;
```

Comment:
Have you tested this on Windows? UtfConverterWide is defined as std::u32string here, while for non-Win it's a std::wstring_convert.

This part has already been improved and it's better not to include such changes in this patch.